### ISSUE TITLE
Moving to gtag.js

### ISSUE THREAD

**Issue by UndefinedOffset** [axllent/silverstripe-analytics-js#9]
It looks like Google is starting to push users over to gtag.js instead of analytics.js, would you be open to a pull request to move this module over to it as well or would you rather we (WBG) fork the module and brand it as a new module under our org?

**Comment by axllent** [comment]
Hi Ed. This has been on my radar for some time now, and I've been tossing up between updating this module or creating separate project entirely. The issue I have with updating this project is that I for instance have a few projects that also track other events in separate JS scripts, and so switching to gtag.js would effectively break those (major versions aside). Then there is also the current compatibility with Silverstripe 3 (separate branch), would we worry about ss3?

I will be developing these GA modules one way or the other, so I'd prefer no active forked projects. Do you think it would be better for me to stick with this single project and bump the major version (once on gtag), or clone this repo, start a fresh project (eg: axllent/silverstripe-gtag), and then deprecate this one with that as the suggested replacement?

Either way, the new code would likely originate from the current codebase, so I'd really appreciate a PR to get going (I'm a bit short of time at the moment). Open to your feedback/thoughts.

**Comment by UndefinedOffset** [comment]
Hmm ya I went back and forth as well with whether it would just be a major version bump or a new project (primarily thinking from the pull request perspective). I think you could really do either a major version bump or a new project which I know doesn't help 😉.

Honestly I think if it was me I'd probably go with a major version bump updating documentation, and descriptions as you go of course. Making it clear that the major version bump swaps out analytics.js for gtag.js. I think the package name is generic enough it fits fine for a version bump. You may indeed see a pull request in the next week or so depending if/when I get a hole in my schedule for it, we've (WBG) been using this module for quite a while on several projects and a couple have received the email Google has been sending out talking about upgrading to gtag.

I was looking at the differences between the two and it seems like they're pretty similar https://developers.google.com/analytics/devguides/collection/upgrade/analyticsjs. One thing that isn't overly clear to me is how to get it multiple tracking keys working, it [kind of sounds like](https://developers.google.com/analytics/devguides/collection/gtagjs/pages) you just add calls to `gtag('config', 'GA_MEASUREMENT_ID_1')` optionally assigning groups and maybe including the script for each that I'm not clear on. They've also seemingly removed tracker names from what I can see.

**Comment by axllent** [comment]
Sweet, then I will keep the current codebase, create a branch for the analytics.js code (ie: v3 stuff), and we can switch to a version 4 release tags once done. I really appreciate your input (and your SortableGridfield module - used that on countless sites too!).

It does look like you can use [multiple tracking keys](https://developers.google.com/analytics/devguides/collection/gtagjs/pages#multiple_properties) - though testing will tell I guess how that all fits in. Obviously the outgoing/download/email link-tracking JS code would need to be adapted to track events, however I don't think it'll be a huge change. Also I do not think the Cache controller ever had any benefit, as ad-blockers still blocked the track itself (the purpose I built it in the first place), so no objection to removing that "feature" ;-) If you have any questions then please just yell out.

**Comment by axllent** [comment]
@UndefinedOffset Thanks again so much for the contribution! I merged your branch (after a few tests), and then made a series of changes to yours, mainly around the JS event tracking which wasn't working. I've pushed all my changes through to `master` and just thought I'd run it by you first in case you had already come across some things I have "fixed" and intentionally done them differently? Everything I've tested appears to work as expected including multiple trackers and even tracking (a single `ga()` event track will push distribute to all trackers, so does not need to be run/specified individually for each).
